src/battle_setup.c

```diff
@@ -1349,15 +1350,42 @@ void BattleSetup_StartTrainerBattle(void)
     ScriptContext_Stop();
 }
 
+static void HandleBattleVariantParty(void)
+{
+    if (FlagGet(B_FLAG_SKY_BATTLE))
+    {
+        SaveChangesToPlayerParty();
+        LoadPlayerParty(); 
+        FlagClear(B_FLAG_SKY_BATTLE);
+    }
+}
+
+static bool8 PlayerHasHealthyPokemon(void){
```

Comment:
I feel like there _should_ be a function which already does that.

Reply:
We could check if (GetMonsStateToDoubles_2 > 0) instead. I'm having trouble navigating the battle code to figure out how that determines when its time to white out, I'm curious what that uses.

Reply:
nvm NoAliveMonsForPlayer works too!